lumen/sources/base.py

```diff
@@ -122,10 +122,10 @@ def wrapped(self, table: str | None = None, limit: int | None = None):
             with lock:
                 with main_lock:
                     new_schema = self._get_schema_cache() or {}
-                if missing in new_schema and limit is None:
+                if missing in new_schema and limit is None and shuffle is False:
```

Comment:
Why not?
```suggestion
                if missing in new_schema and limit is None and not shuffle:
```

Same above.